packages/extension-ui/src/messaging.ts

```diff
@@ -152,8 +153,21 @@ export async function getMetadata (genesisHash?: string | null, isPartial = fals
   }
 
   const def = await request;
-
   if (def) {
+    const specVersionInState = def.specVersion;
```

Comment:
does it make sense to persist metadata we've already fetched? 💭

Reply:
yes... we are storing it once fetched at line 168 - await sendMessage('pri(metadata.set)', def);
this will persist it for next usage..

Reply:
so if we change networks and back again will it fetch 2 times or 1 time?

Reply:
1 time.